### PR TITLE
Rfc 1 align package npm and map commands

### PR DESCRIPTION
**Pull request by digitalsadhu**
Implements the changes outlined in [RFC 1](https://github.com/eik-lib/issues/issues/1) and [RFC 2](https://github.com/eik-lib/issues/issues/2)

A good part of RFC 2 involved using the changes to Eik Config in the `@eik/common` next branch.
 
Additionally:
* Reworks and cleans up the programmatic api (since this PR is already very breaking)
* increase test coverage

### CODE REVIEW

**Comment by trygve-lie** [comment]
I set this one to be merged into the `next` branch for a pre-release.

File: commands/init.js [hunk 1]
```diff
@@ -61,18 +63,22 @@ exports.handler = async (argv) => {
...
             // noop
         }
-        
+
         if (assetFileExists) {
-            throw new Error(`An "eik.json" file already exists in directory. File will not be written`);
+            throw new Error(
+                `An "eik.json" file already exists in directory. File will not be written`,
+            );
         }
```
Comment (gingermusketeer) [inline commands/init.js:82]: Unrelated to this PR but perhaps this command should make use of the config store util?

**Review by gingermusketeer — COMMENTED** [review]
Looks like the pattern is to use the config store to load the config in the `commands` and then unpack the config before passing it down to `classes` where a config object is initialised again. Any reason for this? 

If using the classes without a config object is desired would it be better to have a `.fromOptions` method on the class that sets up the config and  passes it into the constructor?

As an aside it might be easier to grok this PR if the styling changes was a separate PR?

File: commands/init.js (hunk 1 above)
Comment (digitalsadhu) [reply commands/init.js:82]: yea could be

**Comment by digitalsadhu** [comment]
Yea a .fromOptions could be nice way to go

**Review by trygve-lie — APPROVED** [review]
This is a bit big to properly review in go. I'm fine with this. Lets do a `next` release :+1:

**Comment by digitalsadhu** [comment]
Yea sorry about that, I ended up back and forwards between this and the common package and it got a bit out of control. Will endeavour to keep them smaller in future.

**Pull request merged**